projects/storefrontlib/src/lib/checkout/guards/shipping-address-set.guard.ts

```diff
@@ -0,0 +1,38 @@
+import { Injectable } from '@angular/core';
+import { CanActivate, Router, UrlTree } from '@angular/router';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+import { CheckoutDetailsService } from '../checkout-details.service';
+import { CheckoutStep } from '../config/model/checkout-step.model';
+import { CheckoutConfig } from '../config/checkout-config';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ShippingAddressSetGuard implements CanActivate {
+  constructor(
+    private checkoutDetailsService: CheckoutDetailsService,
+    private router: Router,
+    private config: CheckoutConfig
+  ) {}
+
+  canActivate(): Observable<boolean | UrlTree> {
+    const route = this.config.checkout.steps.find(
+      (step: CheckoutStep) => step.type.indexOf('shippingAddress') !== -1
+    );
+    if (!route) {
+      console.warn('not provided route for shippingAddress');
```

Comment:
I would change this message to:
`Missing step with type `shippingAddress` in checkout configuration.`